server/aesdsocket.c

```diff
@@ -125,109 +225,89 @@ int main(int argc, char **argv) {
     addr.sin_addr.s_addr = htonl(INADDR_ANY);
 
     if (bind(listenfd, (struct sockaddr*)&addr, sizeof(addr)) < 0) {
-        syslog(LOG_ERR, "bind: %m");
+        syslog(LOG_ERR, "bind() failed with errno=%d (%s)", errno, strerror(errno));
         close(listenfd);
         closelog();
-        return -1;
+        return EXIT_FAILURE;
     }
     if (listen(listenfd, BACKLOG) < 0) {
-        syslog(LOG_ERR, "listen: %m");
+        syslog(LOG_ERR, "listen() failed with errno=%d (%s)", errno, strerror(errno));
         close(listenfd);
         closelog();
-        return -1;
+        return EXIT_FAILURE;
     }
 
     if (opt_daemon) daemonize();
 
+    {
+        int fd = open(DATAFILE, O_WRONLY | O_CREAT | O_TRUNC, 0664);
+        if (fd < 0) {
+            syslog(LOG_ERR, "Failed to open %s at startup: %s", DATAFILE, strerror(errno));
+        } else {
+            close(fd);
+        }
+    }
+
+    pthread_t ts_tid;
+    pthread_create(&ts_tid, NULL, timestamp_thread, NULL);
+
     while (!g_exit) {
         struct sockaddr_in cli;
         socklen_t clilen = sizeof(cli);
         int connfd = accept(listenfd, (struct sockaddr*)&cli, &clilen);
         if (connfd < 0) {
             if (errno == EINTR && g_exit) break;
             if (errno == EINTR) continue;
-            syslog(LOG_ERR, "accept: %m");
-            break;
+            if (g_exit) break;
+            syslog(LOG_ERR, "accept() failed with errno=%d (%s)", errno, strerror(errno));
+            continue;
         }
 
-        char ip[INET_ADDRSTRLEN] = "unknown";
-        (void)inet_ntop(AF_INET, &cli.sin_addr, ip, sizeof ip);
-        syslog(LOG_INFO, "Accepted connection from %s", ip);
+        int *fd_arg = malloc(sizeof(int));
+        *fd_arg = connfd;
 
-        char *pkt = NULL;
-        size_t cap = 0, len = 0;
-        int have_newline = 0;
-        char buf[RECV_CHUNK];
+        pthread_t tid;
+        pthread_create(&tid, NULL, handle_client, fd_arg);
 
-        while (!g_exit && !have_newline) {
-            ssize_t n = recv(connfd, buf, sizeof buf, 0);
-            if (n == 0) {
-                break; // peer closed
-            }
-            if (n < 0) {
-                if (errno == EINTR) continue;
-                break;
-            }
+        struct thread_node *node = malloc(sizeof *node);
+        node->tid = tid;
+        node->client_fd = connfd;
+        pthread_mutex_lock(&list_mutex);
```

Comment:
Since `g_thread_list` is only accessed from a single thread I don't think mutex locking should be necessary?